src/main/java/com/github/jhonnymertz/wkhtmltopdf/wrapper/Pdf.java

```diff
@@ -55,6 +55,12 @@ public class Pdf {
      */
     private int timeout = 10;
 
+    /**
+     * wkHTMLtoPDF uses a rather old version of WebKit, which often has trouble with up-to-date JavaScript code.
+     * When using window-status and a SyntaxError: Parse error occurs in JavaScript, it can result in the failure of setting the window-status. The wkhtmltopdf process will remain in a waiting state indefinitely. In such cases, a timeout is needed to handle this situation.
+     * */
+    private int windowStatusTimeout = 30;
```

Comment:
I'm very sorry, I had a busy day at work today, and I now have time to respond. 🙏🙏🙏

Regarding `windowStatusTimeout`, I think I need to explain the scenario in which this issue arises:

When you add the `--window-status` parameter in the command, wkhtmltopdf considers the current page to be loaded only after you have completed the assignment `window.status = "complete"` in JavaScript. It typically occurs in situations where the page content is generated asynchronously. Failure to add this parameter can result in incomplete rendering of the page content.

If you have added this parameter in the command but, for some reason, the page has not completed the assignment, wkhtmltopdf will think that the current page has not loaded and will continue to wait. This is the reason why `process.waitFor()` is blocked.

This issue is often caused by developers unfamiliar with the features of wkhtmltopdf. When new features or dependencies are added, developers may use ES6 or syntax not supported by wkhtmltopdf's WebKit, leading to a `SyntaxError: Parse error` warning during the parsing of JavaScript. This interruption in normal JavaScript execution prevents the completion of the assignment. Since it is a warning rather than an error, wkhtmltopdf does not exit but continues to wait.

Therefore, I believe it is a timeout caused by the `window.status` parameter, but it does fall within the range of a runtime timeout.

Regarding the 30s, in our use case, there are often concurrent processing situations, and 10s may not be sufficient. However, it is configurable, so I think there is no need to modify it.

You are right, and I will remove this parameter. I apologize for any confusion it may have caused you. 🙏🙏🙏

My English is not very good; it was generated by OpenAI's translation. Please forgive any inaccuracies.

Reply:
Hey @xweiba no need to apologize! I'm from Brazil, so my English is not good neither =D
Anyway, I really appreciate your contribution and explanation of the issue. Now I fully understand the issue and I'm confident we can merge it. Thank you for this!